pkg/istio-agent/xds_proxy.go

```diff
@@ -272,10 +273,19 @@ func (p *XdsProxy) HandleUpstream(ctx context.Context, con *ProxyConnection, xds
 	upstream, err := xds.StreamAggregatedResources(ctx,
 		grpc.MaxCallRecvMsgSize(defaultClientMaxReceiveMessageSize))
 	if err != nil {
-		proxyLog.Errorf("failed to create upstream grpc client: %v", err)
+		p.connectedMutex.Lock()
```

Comment:
Could just use atomic.Int32? might be a bit simpler

Reply:
sure